lib/asciidoctor/syntax_highlighter.rb

```diff
@@ -0,0 +1,82 @@
+module Asciidoctor
+  module SyntaxHighlighter
```

Comment:
I'm wondering where we should put the documentation of `highlight` since we don't have an Interface or an Abstract class. Should we need to define something like:

```rb
module ServerSideSyntaxHighlighter 
  # documentation goes here
  def highlight node, source, language, opts
    fail NotImplementedError, "A server-side syntax highlighter must implement #highlight"
  end
end
```

Reply:
I considered having a separate base class for a build-time highlighter. It's a superset of a client-side highlighter.

Reply:
> Maybe it's too early and the API is still evolving but at some point we should definitely add documentation on this class.

Of course. It's just taken me all the time I've had up to this point just to get this working. So API documentation is a natural next step, after the tests.

Reply:
API docs added!

Reply:
I also added a NotImplementedError as you suggested in case it is called out of bounds.